### ISSUE TITLE
`RUF031` (`incorrectly-parenthesized-tuple-in-subscript`) - false positive on types

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#12773]
```toml
# ruff.toml
[lint.ruff]
parenthesize-tuple-in-subscript = true
```
```py
foo = dict[str, str] = {} # error: Use parentheses for tuples in subscripts.
```

subscripts have a completely different meaning in type annotations, and the fact that the arguments to `dict` are a tuple is an implementation detail of the syntax, so i don't think this rule should apply to them.

**Comment by dhruvmanila** [comment]
I think this makes sense. I wonder if generics should be accounted here as well. For example:
```py
class Foo(Generic[T1, T2]):
	...
```

@AlexWaygood @MichaReiser Any thoughts?

**Comment by DetachHead** [comment]
the same applies in that case too imo, because the subscript syntax is being used to represent generics

**Comment by dhruvmanila** [comment]
Oh, I think this is covered by https://github.com/astral-sh/ruff/issues/12758 and a fix for it is already up https://github.com/astral-sh/ruff/pull/12762.

**Comment by dhruvmanila** [comment]
(Merging this with https://github.com/astral-sh/ruff/issues/12758)
